spec/unit/plugins/linux/network_spec.rb

```diff
@@ -340,6 +378,15 @@
         expect(plugin['network']['interfaces'].keys.sort).to eq(["eth0", "eth0.11", "eth0.151", "eth0.152", "eth0.153", "eth0:5", "eth3", "foo:veth0@eth0", "lo", "ovs-system",  "tun0", "venet0", "venet0:0", "xapi1"])
       end
 
+      it "detects the layer one details of an ethernet interface" do
+        expect(plugin['network']['interfaces']['eth0']['link_speed']).to eq(10000)
+        expect(plugin['network']['interfaces']['eth0']['duplex']).to eq('Full')
+        expect(plugin['network']['interfaces']['eth0']['port']).to eq('FIBRE')
+        expect(plugin['network']['interfaces']['eth0']['transceiver']).to eq('external')
+        expect(plugin['network']['interfaces']['eth0']['auto-negotiation']).to eq('on')
+        expect(plugin['network']['interfaces']['eth0']['mdi-x']).to be_nil
```

Comment:
`auto-negotiation` and `mdi-x` should use underscores instead of hyphens to be compatible with symbols.